static/js/settings_org.js

```diff
@@ -935,85 +935,63 @@ exports.build_page = function () {
-            cache: false,
-            processData: false,
-            contentType: false,
-            success: function () {
-                loading.destroy_indicator($("#upload_icon_spinner"));
-                $("#upload_icon_button_text").expectOne().show();
-            },
-            error: function (xhr) {
-                loading.destroy_indicator($("#upload_icon_spinner"));
-                $("#upload_icon_button_text").expectOne().show();
-                ui_report.error("", xhr, error_field);
-            },
-        });
-
-    }
-    realm_icon.build_realm_icon_widget(upload_realm_icon);
-
-    function upload_realm_logo(file_input, night) {
+    function upload_realm_logo_or_icon(file_input, night, icon) {
         const form_data = new FormData();
         let spinner;
         let error_field;
         let button_text;
+        let url;
 
         form_data.append('csrfmiddlewaretoken', csrf_token);
         for (const [i, file] of Array.prototype.entries.call(file_input[0].files)) {
             form_data.append('file-' + i, file);
         }
-        if (night) {
-            error_field = $("#night-logo-section .realm-logo-file-input-error");
-            spinner = $("#night-logo-section .upload-logo-spinner");
-            button_text = $("#night-logo-section .upload-logo-button-text");
+        if (icon) {
+            url = '/json/realm/icon';
+            error_field = $("#realm_icon_file_input_error");
+            spinner = $("#upload_icon_spinner").expectOne();
+            loading.make_indicator(spinner, {text: i18n.t("Uploading profile picture.")});
+            button_text = $("#upload_icon_button_text");
         } else {
-            error_field = $("#day-logo-section .realm-logo-file-input-error");
-            spinner = $("#day-logo-section .upload-logo-spinner");
-            button_text = $("#day-logo-section .upload-logo-button-text");
+            if (night) {
+                error_field = $("#night-logo-section .realm-logo-file-input-error");
+                spinner = $("#night-logo-section .upload-logo-spinner");
+                button_text = $("#night-logo-section .upload-logo-button-text");
+            } else {
+                error_field = $("#day-logo-section .realm-logo-file-input-error");
+                spinner = $("#day-logo-section .upload-logo-spinner");
+                button_text = $("#day-logo-section .upload-logo-button-text");
+            }
+            url = '/json/realm/logo';
+            spinner.expectOne();
+            loading.make_indicator(spinner, {text: i18n.t("Uploading logo.")});
+            form_data.append('night', JSON.stringify(night));
         }
-        spinner.expectOne();
+        button_text.expectOne();
         error_field.hide();
-        button_text.expectOne().hide();
-        loading.make_indicator(spinner, {text: i18n.t("Uploading logo.")});
-        form_data.append('night', JSON.stringify(night));
+        button_text.hide();
         channel.post({
```

Comment:
It's not clear to me why hide is separate, can't it be the same way as it was before as `button_text.expectOne().hide()`

Reply:
ok